### PR TITLE
Add SBOM guidelines

### PR DESCRIPTION
**Pull request by mprpic**
Extensive documentation of the structure of Red Hat's SBOMs.

### CODE REVIEW

**Comment by mprpic** [comment]
@twaugh I still need to finish documenting the exact fields for the different component types, but the intro should be ready for review. I'd like to make sure we have all of the SBOM types documented clearly.

File: docs/sbom.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+process itself. This SBOM type also aligns with the _Build_ SBOM type from CISA's guidance on
+[Types of SBOM Documents](https://www.cisa.gov/sites/default/files/2023-04/sbom-types-document-508c.pdf).
+
+_Release-time_ SBOMs are generated when an artifact is released or published. These SBOMs build upon build-time
+SBOMs by incorporating additional metadata, such as the repositories or locations where the artifact is
+published, and associating it with the relevant product information if there is any. Release-time SBOMs reflect the
+state of the software as it is distributed to end users. This SBOM type is close to the _Deployed_ type as defined
+by CISA, but it reflects the state of the product that _would_ be installed by a given end user.
+
+Red Hat's publicly available SBOMs are of the "release" type, including details about where an artifact
```
Comment (twaugh) [inline docs/sbom.md:65]:
```suggestion
Red Hat's publicly available SBOMs are of the "release-time" type, including details about where an artifact
```

File: docs/sbom.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+```diff
+--- sbom/examples/container_image/build/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json
++++ sbom/examples/container_image/release/ubi9-micro-container-9.4-6.1716471860_amd64.spdx.json
+@@ -22,12 +22,12 @@
+         {
+           "referenceCategory": "PACKAGE-MANAGER",
+           "referenceType": "purl",
+-          "referenceLocator": "pkg:oci/ubi-micro@sha256:1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=amd64"
++          "referenceLocator": "pkg:oci/ubi-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=amd64&repository_url=registry.access.redhat.com/ubi9/ubi-micro&tag=9.4-6.1716471860"
```
Comment (twaugh) [inline docs/sbom.md:86]: Another interesting difference here is that `:` gets percent-encoded to `%3A`. Should we be doing that to start with, or we don't care?

File: docs/sbom.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+### Shallow vs Complete
+
+_Shallow_ SBOMs describe only the current layer of components that were used to build a specific artifact. They 
+focus on the immediate components without necessarily documenting their origins. The benefit of this approach is 
+smaller SBOM files, and the ability to make incremental changes in select SBOMs while others remain unchanged when 
+composing multiple SBOMs into a larger set (to describe a product, for example).
+
+_Complete_ SBOMs provide a full, in-depth representation of all levels of the build process in a single file. A
+Complete SBOM includes all component layer, their dependencies, offering a complete view of a product's composition.
```
Comment (twaugh) [inline docs/sbom.md:128]:
```suggestion
Complete SBOM includes all component layers and their dependencies, offering a complete view of a product's composition.
```

**Review by twaugh — COMMENTED** [review]
What you have so far looks great!

File: docs/sbom.md (hunk 2 above)
Comment (mprpic) [reply docs/sbom.md:86]: Yea, we should percent-encode as per spec: https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component

**Pull request merged**